code.js

```diff
@@ -0,0 +1,159 @@
+// Kane Kriz
+// UWYO COSC 3020 Algorithms - parallel mergesort primary code.js file
+// 11 May 2025
+//
+//
+
+
+
+////
+
+
+
+
+function parallelMergeSort(inputArray) 
+{
+    var corePlaceholderCount;
+    var partitionSize;
+    var partitions;
+    var sortedPartitions;
+    var mergedPartitions;
+
+    var inputLength = inputArray.length;
+    corePlaceholderCount = 2;
+
+    if(inputLength <= 1) 
+    {
+        return inputArray;
+    }
+
+    partitionSize = Math.ceil(inputLength / corePlaceholderCount);
+
+    partitions = [];
+  
+    for(var i = 0; i < inputLength; i += partitionSize)
+    {
+        var endPos = i + partitionSize;
+        
+        if(endPos > inputLength)
+        {
+            endPos = inputLength;
+        }
+        
+        partitions.push(inputArray.slice(i, endPos));
+    }
+
+    sortedPartitions = partitions.map(solvePartition);
```

Comment:
The built-in map and reduce functions don't parallelize anything. You can use e.g. the map reduce library we used for the example code in class.